sources.sh

```diff
@@ -254,15 +270,26 @@ jq <<<"$sources" --argjson pins "$externalPinsJson" '
 	# TODO a lot of this could be removed/parsed during the above reduce, since it has to parse things in build order anyhow
 	# TODO actually, instead, this bit should be a totally separate script so the use case of "combine sources.json files together" works better 👀
 	| (
-		# TODO make this faster, this reduce takes the longest time now
-		reduce to_entries[] as $e ({};
-			$e.key as $sourceId
-			| .[ $e.value.arches[] | .tags[], .archTags[] ] |= (
-				.[$e.value.arches | keys[]] |= (
-					. + [$sourceId] | unique_unsorted
-				)
+		# creating a lookup of .[tag][arch] to a list of sourceIds
+		[
+			.[] as $e
```

Comment:
Can we replace `$e` with `$s`?  When I see `$e` my brain says "entry" but this is a "source" object :innocent: